docs: Improve `no-child-traversal-in-connectedcallback` (#130)
43081j/eslint-plugin-wc: docs/rules/no-child-traversal-in-connectedcallback.md

## When Not To Use It

If you are comfortable with the edge cases of DOM traversal directly in the `connectedCallback` then you can disable this rule.

> If your element is appended to the DOM via `.innerHTML` or similar methods, you should disable this rule because in such a case, the children are are available during `connectedCallback` and no mutations events will fire.

Reviewer: I don't think we should add this because it's typically not possible to know what happens to your element, as it is up to the consumer how it is embedded.

For example if your element is appended to the DOM using `append()`, and you have disabled this rule, your code breaks. If your element is served from the source HTML, and browser decides to render, your element is connected but incomplete, you code breaks.

The point of this rule is to remind you that those cases exist, and to carefully think about how you want to approach them.

Author: I think we could at least hint somewhere in the rule text that depending on the renderer, children may be available or not and I think any universally usable component that deals withs children has to check for both immediately and delayed availability of children.

I just fear that people who blindly follow this rule will create components that can not be inserted via `.innerHTML`, which I think is very common with Frameworks like Vue or React.

> If your element is appended to the DOM via `.innerHTML` or similar methods, you may need to disable this rule because in such a case, the children are are available during `connectedCallback` and no mutations events will fire.

Author: ```suggestion
If your element is appended to the DOM via `.innerHTML` or similar methods, you should disable this rule because in such a case, the children are are available during `connectedCallback` and no mutations events will fire.
```

> It may also be necessary to disable the rule when the custom element is appended to the DOM via JavaScript like for instance the Vue framework, which appends the element including all children at once and a `MutationObserver` would not receive any `childList` mutations in such cases.

Author: ```suggestion
If your element is appended to the DOM via `.innerHTML` or similar methods, you may need to disable this rule because in such a case, the children are are available during `connectedCallback` and no mutations events will fire.
```
